app.js

```diff
@@ -331,7 +331,7 @@ const loadTcpServer = net.createServer(function (socket) {
     const freeLoad = availableWorkingCpus - currentLoad
     let freeLoadPercentage = Math.round((freeLoad / availableWorkingCpus) * 100)
     if (freeLoadPercentage <= 0) {
-      freeLoadPercentage = 1 // when its 0 the server is set to drain and will move projects to different servers
+      freeLoadPercentage = 0 // when its 0 the server is set to drain and will move projects to different servers
```

Comment:
WDYT about keeping the `currentLoad` as is and adjust the edge case to
```suggestion
      // when its 0 the server is set to drain and will move projects to different servers
      freeLoadPercentage = os.loadavg()[1] > availableWorkingCpus ? 0 : 1
```
(Maybe store `os.loadavg()` in a variable to save a sys call under heavy load.)

Reply:
I'm missing the advantage of this. Wouldn't it mean we only have 2 values returned, 0 & 1?

Reply:
This is the edge case handling for overloaded CPUs (load1 of 50 for a 5 core box). The proposed change would keep overloaded instances in rotation (with weight 1) for about 5min until the load5 hits the capacity limit.

On a 16 core box:

time in s | load1 | load5 | weight
--- | --- | --- | ---
0 | 0 | 0 | 100
30 | 5 | 0 | 66
60 | 16 | 5 | 1
300 | 16 | 16 | 0

Reply:
(Delaying the drain mode (weight=0) for 5min might stop a stamped of projects in a bad state.)

Reply:
Right, so you were suggesting load5 for going to 0 capacity and load1 for looking moving us to 1. It's a good idea, for the moment I think I want to keep it a bit simpler and just use the single load value, in the next iteration I might look at this. Cheers.